students/kulon_aleksander/lesson_12/validated_user_base.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import argparse
+import csv
+import logging
+import sys
+
+from lesson_08.email_validator\
+    import email_validate as is_valid_email
+from lesson_08.phone_number_validator\
+    import validate_phone_number as is_valid_phone_number
+from lesson_08.postal_code_validator\
+    import validate_postal_code_pl as is_valid_postal_code
+from lesson_08.strong_password_detection\
+    import password_checker as is_strong_password
+
+logging.basicConfig(level=logging.INFO,
+                    format=' %(asctime)s - %(levelname)s - %(message)s')
+
+
+def combined_validator(email, password, phone_number, postal_code):
+    logging.info("Verifying user data")
+
+    if is_valid_email(email) and is_strong_password(password) \
+            and is_valid_phone_number(phone_number) \
+            and is_valid_postal_code(postal_code):
+        logging.info("Verification passed")
+    else:
+        logging.warning("Verification failed")
+        raise Exception("One of arguments didn't pass verification.")
```

Comment:
You can consider implementing your own exception (before line 21), have a look at this:

class ValidateUserBaseException(Exception):
    module = 'ValidateUserBaseException'

    def __init__(self, message):
        super(ValidateUserBaseException, self).__init__(self.module + ': ' + message)
and then
raise ValidateUserBaseException('Email address is invalid')
Of course it can stay as it's now